utils/s3/client.cc

```diff
@@ -1052,17 +1055,20 @@ class client::do_upload_file : private multipart_upload {
                    std::filesystem::path path,
                    sstring object_name,
                    std::optional<tag> tag,
-                   size_t part_size)
+                   size_t part_size,
+                   upload_progress& up)
         : multipart_upload(std::move(cln), std::move(object_name), std::move(tag))
         , _path{std::move(path)}
         , _part_size(part_size)
+        , _progress(up)
     {
     }
 
     future<> upload() {
         auto f = co_await open_file_dma(_path.native(), open_flags::ro);
         const auto stat = co_await f.stat();
         const uint64_t file_size = stat.st_size;
+        _progress.total += file_size;
```

Comment:
If api users poll for the upload task status,  they may see: `99/100` -> `99/500`.
If this is desirable, maybe it's worth mentioning in progress_unit that job size may change?

Reply:
> If api users poll for the upload task status, they may see: `99/100` -> `99/500`.

Yes

> If this is desirable, maybe it's worth mentioning in progress_unit that job size may change?

You mean smth like

```patch
-    _status.progress_units = "bytes";
+    _status.progress_units = "bytes ('total' may grow along the way)";
```

?

Reply:
Yes, so api users are aware

Reply:
OK, updated